modules/services/window-managers/river/default.nix

```diff
@@ -0,0 +1,225 @@
+
+  inputsSettings = let
+    inputSettings = mapAttrsToList (inputName: inputConfig:
+      mapAttrsToList (option: value:
+        let
+          # fooBarOption -> foo-bar-option
+          formatOptionName =
+            builtins.replaceStrings upperChars (map (c: "-${c}") lowerChars);
+
+        in "riverctl input ${inputName} ${formatOptionName option} ${
+          toString value
+        }") (filterAttrs (n: v: v != null) inputConfig)) cfg.config.inputs;
+  in concatStringsSep "\n" (lists.flatten inputSettings);
+
+  inputs = ''
+    ${keyboardLayout}
+
+    ${inputsSettings}
+  '';
+
+  # CONFIGURATION
+  convertColor = colorString: "0x${replaceStrings [ "#" ] [ "" ] colorString}";
+  xCursorTheme = let
+    themeName = cfg.config.xCursorTheme.name;
+    size = cfg.config.xCursorTheme.size;
+  in optionalString (themeName != "") "riverctl xcursor-theme ${themeName}${
+    optionalString (size != null) " ${toString size}"
+  }";
+
+  configuration = ''
+    riverctl attach-mode ${cfg.config.attachMode}
+    riverctl background-color ${convertColor cfg.config.backgroundColor}
+    riverctl border-color-focused ${
+      convertColor cfg.config.border.color.focused
+    }
+    riverctl border-color-unfocused ${
+      convertColor cfg.config.border.color.unfocused
+    }
+    riverctl border-color-urgent ${convertColor cfg.config.border.color.urgent}
+    riverctl border-width ${toString cfg.config.border.width}
+    riverctl focus-follows-cursor ${cfg.config.focusFollowsCursor}
+    riverctl hide-cursor timeout ${toString cfg.config.hideCursor.timeout}
+    riverctl hide-cursor timeout ${
+      if cfg.config.hideCursor.whenTyping then "enabled" else "disabled"
+    }
+    riverctl set-cursor-warp ${cfg.config.cursorWarp}
+    riverctl set-repeat ${cfg.config.repeatRate}
+    ${xCursorTheme}
+  '';
+
+  formatStartupProgram = command: ''riverctl spawn "${command}"'';
+
+in {
+  meta.maintainers = [ maintainers.GaetanLepage ];
+
+  options = import ./options.nix { inherit pkgs lib; };
+
+  config = mkIf cfg.enable {
+    assertions = [
+      (hm.assertions.assertPlatform "wayland.windowManager.river" pkgs
+        platforms.linux)
+    ];
+
+    home.packages = optional (cfg.package != null) cfg.package
+      ++ optional cfg.xwayland pkgs.xwayland;
+
+    xdg.configFile."river/init".source = pkgs.writeShellScript "init" ''
+      ### This file was generated with Nix. Don't modify this file directly.
+
+      ### SHELL VARIABLES ###
+      ${config.lib.shell.exportAll cfg.extraSessionVariables}
+
+      ### RULES ###
+      ${rules}
+
+      ### MAPPINGS ###
+      ${mappings}
+
+      ### CONFIGURATION ###
+      ${configuration}
+
+      ### INPUTS ###
+      ${inputs}
+
+      ### EXTRA CONFIGURATION ###
+      ${cfg.extraConfig}
+
+      ### STARTUP PROGRAMS ###
+      ${concatMapStringsSep "\n" formatStartupProgram cfg.startupPrograms}
+
+      ### LAYOUT ###
+      riverctl default-layout ${cfg.config.layoutGenerator.name}
```

Comment:
Note, the name of the layout is not guaranteed to be equal to the filename of the executable, see <https://github.com/riverwm/river/wiki#my-windows-are-not-tiled>.

Reply:
Thanks for this remark !

Reply:
I have created an option to eventually specify a name for the executable if it is not the same as the layout generator itself.